lib/jekyll_pages_api/page.rb

```diff
@@ -3,14 +3,17 @@
 module JekyllPagesApi
   # wrapper for a Jekyll::Page
   class Page
+    HTML_EXTS = %w(.html .md .markdown .textile)
     attr_reader :page
 
-    def initialize(page)
+    def initialize(page, site)
       @page = page
+      @site = site
     end
 
     def html?
-      %w(.html .md).include?(self.page.ext)
+      path = @page.path
```

Comment:
My main gripe is that if you misspell them, they give back `nil` instead of raising an error, which has caused weird bugs for me in the past. Also a bonus that you can easily swap the reader out for a method without changes anywhere else.